Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs

```diff
@@ -3428,28 +3433,59 @@ private void SwitchFrame(ElementInfo EI)
             }
         }
 
-        public IWebElement LocateElementByLocators(ObservableList<ElementLocator> Locators, ElementInfo currentPOMElementInfo,bool IscallfromFriendlyLocator = false)
+        public IWebElement LocateElementByLocators(ElementInfo currentPOMElementInfo,bool iscallfromFriendlyLocator=false, POMExecutionUtils POMExecutionUtils=null)
         {
             IWebElement elem = null;
-            foreach (ElementLocator locator in Locators)
+            foreach (ElementLocator locator in currentPOMElementInfo.Locators)
             {
                 locator.StatusError = string.Empty;
                 locator.LocateStatus = ElementLocator.eLocateStatus.Pending;
             }
 
-            foreach (ElementLocator locator in Locators.Where(x => x.Active == true).ToList())
+            foreach (ElementLocator locator in currentPOMElementInfo.Locators.Where(x => x.Active == true).ToList())
             {
-                if (!locator.IsAutoLearned)
+                List<FriendlyLocatorElement> friendlyLocatorElementlist = new List<FriendlyLocatorElement>();
+                if (locator.EnableFriendlyLocator && !iscallfromFriendlyLocator)
                 {
-                    elem = LocateElementIfNotAutoLeared(locator);
+                    IWebElement targetElement = null;
+                   
+                    foreach (ElementLocator FLocator in currentPOMElementInfo.FriendlyLocators.Where(x=>x.Active == true).ToList())
```

Comment:
move the code of friendly locator if block to a new function to make it more modeular

Reply:
Their are only 3 line code will check how to convert it into function it will handled in next PR